lib/datasource/deb/index.ts

```diff
@@ -0,0 +1,306 @@
+   * - release: buster, etc.
+   */
+  override readonly defaultRegistryUrls = [
+    'https://ftp.debian.org/debian?suite=stable&components=main,contrib,non-free',
+  ];
+
+  override readonly defaultConfig: DebLanguageConfig = {
+    deb: {
+      binaryArch: 'amd64',
+      downloadDirectory: './others/deb/download',
+      extractionDirectory: './others/deb/extracted',
+    },
+  };
+
+  override readonly caching = false; // TODO: how can this be used?
+
+  /**
+   * Here, we tell Renovate that this data source can respect multiple upstream repositories
+   */
+  override readonly registryStrategy = 'merge';
+
+  /**
+   * Not all Debian packages follow Semver, so it's wise to keep this loose but make sure to
+   * have enough tests in your application.
+   */
+  override readonly defaultVersioning = 'loose';
+
+  requiredPackageKeys = ['Package', 'Version', 'Homepage'];
+
+  async initCacheDir(cfg: DebLanguageConfig): Promise<void> {
+    await fs.ensureCacheDir(cfg.deb.downloadDirectory);
+    await fs.ensureCacheDir(cfg.deb.extractionDirectory);
+  }
+
+  async extract(
+    compressedFile: string,
+    compression: string,
+    outputFile: string
+  ): Promise<void> {
+    if (compression === 'gz') {
+      const source = fs.createReadStream(compressedFile);
+      const destination = fs.createWriteStream(outputFile);
+      await fs.pipeline(source, createUnzip(), destination);
+    } else {
+      throw 'Unknown compression standard, this is probably a programming error';
+    }
+  }
+
+  async downloadAndExtractPackage(
+    cfg: DebLanguageConfig,
+    basePackageUrl: string
+  ): Promise<string> {
+    // we hash the package URL and export the hex to make it file system friendly
+    // we use the hashed url as the filename for the local directories/files
+    const hash = createHash('sha256');
+    hash.update(basePackageUrl);
+    const hashedPackageUrl = hash.copy().digest('hex');
+    const downloadDirectory = await fs.ensureCacheDir(
+      cfg.deb.downloadDirectory
+    );
+
+    const extractionDirectory = await fs.ensureCacheDir(
+      cfg.deb.extractionDirectory
+    );
+    const extractedFile = extractionDirectory + '/' + hashedPackageUrl + '.txt';
+    let extractedFileExists = false;
+    try {
+      await fs.access(extractedFile, fs.constants.R_OK);
+      extractedFileExists = true;
+    } catch (e) {
+      //ignore
+    }
+
+    for (let i = 0; i < DebDatasource.compressions.length; i++) {
+      const compression = DebDatasource.compressions[i];
+      const compressedFile =
+        downloadDirectory + '/' + hashedPackageUrl + '.' + compression;
+      try {
+        const wasUpdated = await this.downloadPackageFile(
+          basePackageUrl,
+          compression,
+          compressedFile
+        );
+        if (wasUpdated || !extractedFileExists) {
+          await this.extract(compressedFile, compression, extractedFile);
```

Comment:
```suggestion
          await this.extract(compressedFile, compression, extractedFile);
          await fs.rm(compressedFile);
```

delete compressed file after extraction

Reply:
Actually, I think can use the timestamp of the extracted file, which should be roughly the same as the downloaded file.

Reply:
I don't think so, it depends on extraction code. It can use embedded metadata in compressed file and set the file timestamp, so it will always older that the compressed file date.